metricbeat/module/redis/keyspace/keyspace_integration_test.go

```diff
@@ -25,54 +25,80 @@ import (
 
 	"github.com/elastic/beats/libbeat/tests/compose"
 	mbtest "github.com/elastic/beats/metricbeat/mb/testing"
-	"github.com/elastic/beats/metricbeat/module/redis"
 
 	rd "github.com/garyburd/redigo/redis"
 	"github.com/stretchr/testify/assert"
 )
 
-var host = redis.GetRedisEnvHost() + ":" + redis.GetRedisEnvPort()
+var runner = compose.TestRunner{
+	Service: "redis",
+	Options: map[string][]string{
+		"REDIS_VERSION": []string{
+			"3.2.4",
+			"4.0.11",
+			"5.0-rc",
+		},
+		"IMAGE_OS": []string{
+			"alpine",
+			"stretch",
+		},
+	},
+	Parallel: true,
+}
 
 func TestFetch(t *testing.T) {
-	compose.EnsureUp(t, "redis")
-
-	addEntry(t)
+	runner.Run(t, func(t *testing.T, host string) {
```

Comment:
Nice idea 👍

Reply:
I am starting to think in the merge of this little monster and one of my main concerns is that this tests structure modifies the whole integration test files even if only for the indentation level. This is already making the maintenance of the branch a bit painful, as any change on integration tests require a manual merge when updating the branch. And for the same reasons it will be also painful when merged for open PRs and backports.

I am thinking on a second option, in which there is a simple test that calls the runner and the actual tests are defined as private functions. Find in [this gist](https://gist.github.com/jsoriano/5012214f5c3578489f1d800d2f0a9aa3) both options for a simple test file and their diffs with master. Option 1 is the one I'm following by now, option 2 would be my new proposal, in option 2 the differences are actually only in the lines that would need to be changed, and there are less indentation levels.

@andrewkroh @ruflin opinions?

Reply:
any disadvantages for option 2 besides having more functions? +1 on 2 so far.

Reply:
The more I think about it, the more I am leaning about second option too. One disadvantage can be that one needs to add the test function to the suite for it to be run, but most of integration tests have only a fetch and a data function and is not so frequent to add more tests.
I'll go for option 2.